src/main/java/org/jitsi/videobridge/ratecontrol/AdaptiveSimulcastBitrateController.java

```diff
@@ -0,0 +1,546 @@
+                OVERSENDING_COEF
+                    = cfg.getDouble(OVERSENDING_COEF_PNAME, OVERSENDING_COEF);
+                OVERSENDING_INTERVAL_MS
+                    = cfg.getInt(
+                        OVERSENDING_INTERVAL_MS_PNAME, OVERSENDING_INTERVAL_MS);
+                INITIAL_PERIOD_MS
+                    = cfg.getInt(INITIAL_PERIOD_MS_PNAME, INITIAL_PERIOD_MS);
+                ENABLE_TRIGGER
+                    = cfg.getBoolean(ENABLE_TRIGGER_PNAME, ENABLE_TRIGGER);
+            }
+
+            recurringProcessibleExecutor = new RecurringProcessibleExecutor();
+            configurationInitialized = true;
+        }
+    }
+
+    /**
+     * The {@link VideoChannel} which owns this {@link
+     * AdaptiveSimulcastBitrateController}.
+     */
+    private final VideoChannel channel;
+
+    /**
+     * The latest estimation of the available bandwidth as reported by our
+     * {@link Channel}'s {@link MediaStream}'s {@link BandwidthEstimator}.
+     */
+    private long latestBwe = -1;
+
+    /**
+     * The time that {@link #process()} was last called.
+     */
+    private long lastUpdateTime = -1;
+
+    /**
+     * Whether this {@link AdaptiveSimulcastBitrateController} has triggered
+     * suppression of the HD layer.
+     */
+    private boolean triggered = false;
+
+    /**
+     * A queue which holds recent data about the sending bitrate and available
+     * bandwidth.
+     */
+    private History history = new History();
+
+    /**
+     * Initializes a new {@link AdaptiveSimulcastBitrateController} instance.
+     *
+     * @param channel the {@link VideoChannel} which owns this instance.
+     * serve.
+     */
+    public AdaptiveSimulcastBitrateController(
+                LastNController lastNController,
+                VideoChannel channel)
+    {
+        this.channel = channel;
+
+        initializeConfiguration(
+                ServiceUtils.getService(
+                        channel.getBundleContext(),
+                        ConfigurationService.class));
+
+        // Create a bandwidth estimator and hook us up to changes to the
+        // estimation.
+        BandwidthEstimator be
+            = ((VideoMediaStream) channel.getStream())
+                .getOrCreateBandwidthEstimator();
+        be.addListener(this);
+
+        recurringProcessibleExecutor.registerRecurringProcessible(this);
+    }
+
+    /**
+     * Releases resources used by this instance and stops the periodic execution
+     * of {@link #process()}.
+     */
+    @Override
+    public void close()
+    {
+        recurringProcessibleExecutor.deRegisterRecurringProcessible(this);
+    }
+
+    /**
+     * Notifies this instance that the estimation of the available bandwidth
+     * has changed.
+     *
+     * @param bwe the new estimation of the available bandwidth in bits per second.
+     */
+    @Override
+    public void bandwidthEstimationChanged(long bwe)
+    {
+        latestBwe = bwe;
```

Comment:
Why do we care about the remote send bandwidth estimation (that's what the bandwidth estimation algorithm running at the bridge is calculating)? We can't use those estimations to figure out if we're oversending, instead we need to take into account the values in the REMBs. Or am I missing something obvious?